### PR TITLE
Multiple changes to Course:

### LINKED ISSUE #242: Time window for course enrollment

### LINKED ISSUE #261: Allow course attributes to be editable
Related to #242

### PR DESCRIPTION
**Pull request by ankitjavalkar**
- Allow a start and end time for Course enrollment | Closes #242 
- Allow Course to be editable | Closes #261

### CODE REVIEW

File: yaksh/views.py [hunk 1]
```diff
@@ -614,10 +614,43 @@ def add_course(request):
...
+
+    if not is_moderator(user):
+        raise Http404('You are not allowed to view this page')
+    if request.method == 'POST':
+        form = CourseForm(request.POST, instance=course)
+        if form.is_valid():
+            new_course = form.save(commit=False)
+            new_course.creator = user
+            new_course.save()
+            return my_redirect('/exam/manage/')
```
Comment (maheshgudi) [inline yaksh/views.py]: could this be redirected to <code>/exam/manage/courses/</code>

File: yaksh/views.py [hunk 2]
```diff
@@ -676,6 +709,11 @@ def enroll(request, course_id, user_id=None, was_rejected=False):
         raise Http404('You are not allowed to view this page')
 
     course = get_object_or_404(Course, pk=course_id)
+    if not course.is_active_enrollment:
+        msg = 'Enrollment for this course has been closed, please contact your '\
+            'instructor/administrator.'
+        return complete(request, msg, attempt_num, questionpaper_id=None)
```
Comment (prathamesh920) [inline yaksh/views.py]: **attempt_num** is not defined

File: yaksh/views.py [hunk 3]
```diff
@@ -614,10 +614,43 @@ def add_course(request):
...
+
+@login_required
 def enroll_request(request, course_id):
     user = request.user
     ci = RequestContext(request)
     course = get_object_or_404(Course, pk=course_id)
+    if not course.is_active_enrollment:
+        msg = 'Enrollment for this course has been closed, please contact your '\
+            'instructor/administrator.'
+        return complete(request, msg, attempt_num, questionpaper_id=None)
```
Comment (prathamesh920) [inline yaksh/views.py]: **attempt_num** is not defined
I think we should redirect the user to same page i.e dashboard, here we do not need to return complete()

File: yaksh/views.py [hunk 4]
```diff
@@ -614,10 +614,43 @@ def add_course(request):
 
 
 @login_required
+def edit_course(request, course_id=None):
```
Comment (prathamesh920) [inline yaksh/views.py]: edit_course and add_course are redundant.
For edit course, we can call add_course with course instance.

File: yaksh/test_models.py [hunk 5]
```diff
@@ -843,6 +843,29 @@ def setUp(self):
         self.quiz1 = Quiz.objects.get(description='demo quiz 1')
         self.quiz2 = Quiz.objects.get(description='demo quiz 2')
 
+        # create courses with
```
Comment (prathamesh920) [inline yaksh/test_models.py]: Is this comment incomplete?

File: yaksh/test_models.py [hunk 6]
```diff
@@ -843,6 +843,29 @@ def setUp(self):
...
+            name="Open Course With Enrollment Disabled",
+            enrollment="Open Course",
+            creator=self.creator,
+            start_enroll_time=datetime(2015, 10, 9, 10, 8, 15, 0,
+                tzinfo=pytz.utc
+            ),
+            end_enroll_time=datetime(2015, 11, 9, 10, 8, 15, 0,
+               tzinfo=pytz.utc
+            ),
+        )
```
Comment (prathamesh920) [inline yaksh/test_models.py]: Should we have tests for enrollment enabled as well?

File: yaksh/forms.py [hunk 7]
```diff
@@ -276,7 +276,8 @@ class CourseForm(forms.ModelForm):
 
     class Meta:
         model = Course
-        fields = ['name', 'active', 'enrollment', 'instructions']
+        fields = ['name', 'active', 'enrollment',
```
Comment (ankitjavalkar) [inline yaksh/forms.py]: Fixed

File: yaksh/views.py (hunk 1 above)
Comment (ankitjavalkar) [inline yaksh/views.py]: Fixed

File: yaksh/views.py (hunk 4 above)
Comment (ankitjavalkar) [inline yaksh/views.py]: Done

File: yaksh/views.py (hunk 3 above)
Comment (ankitjavalkar) [inline yaksh/views.py]: - attempt_num is fixed.
- The ```complete()``` displays the message as an error, the template that is displayed has a button that redirects to dashboard

File: yaksh/views.py (hunk 2 above)
Comment (ankitjavalkar) [inline yaksh/views.py]: Fixed

File: yaksh/test_models.py (hunk 5 above)
Comment (ankitjavalkar) [inline yaksh/test_models.py]: Fixed

File: yaksh/test_models.py (hunk 6 above)
Comment (ankitjavalkar) [inline yaksh/test_models.py]: Done

**Pull request merged**